--- makepath.c.orig	1996-01-01 10:08:22.000000000 +0300
+++ makepath.c	2010-08-19 00:43:46.000000000 +0400
@@ -127,6 +127,11 @@
 	}
 
       slash = dirpath;
+#ifdef __MSDOS__
+      /* Skip the drive letter, no need to create it.  */
+      if (isalpha(slash[0]) && slash[1] == ':')
+	slash += 2;      
+#endif
       while (*slash == '/')
 	slash++;
       while ((slash = index (slash, '/')))
@@ -137,7 +142,7 @@
 #endif
 	  *slash = '\0';
 	  if (stat (dirpath, &stats))
-	    {
+		{
 #ifdef HPUX_CDF
 	      /* If this component of the pathname ends in `+' and is
 		 followed by 2 `/'s, then this is a CDF.  We remove the
